WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift

```diff
@@ -141,34 +151,67 @@ final class UpdateProductInventoryViewModel: ObservableObject {
         inventoryItem.imageURL
     }
 
-    func onTapIncreaseStockQuantityOnce() async {
+    func onTapIncreaseStockQuantityOnce() async throws {
         guard let quantityDecimal = Decimal(string: quantity) else {
             return
         }
 
         let newQuantity = quantityDecimal + 1
         quantity = newQuantity.formatted()
 
-        try? await updateStockQuantity(with: newQuantity)
+        do {
+            try await updateStockQuantity(with: newQuantity)
+        } catch {
+            throw UpdateInventoryError.generic
+        }
     }
 
-    func onTapUpdateStockQuantity() async {
+    func onTapUpdateStockQuantity() async throws {
         guard let quantityDecimal = Decimal(string: quantity) else {
-            return
+            throw UpdateInventoryError.nonSupportedQuantity
         }
 
-        try? await updateStockQuantity(with: quantityDecimal)
+        do {
+            try await updateStockQuantity(with: quantityDecimal)
+        } catch {
+            throw UpdateInventoryError.generic
```

Comment:
Same here: if we throw this error in updateStockQuantity, we don't need to throw it here again